tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/ant/CleanOutputDirectoryTask.java

```diff
@@ -340,4 +338,77 @@ private static ImmutableList<String> readLinesFromResource(String name) {
             throw new RuntimeException("cannot read resource: " + name, e);
         }
     }
+
+    private static Retain getRetain(Element elem) {
+        if (!"retain".equals(elem.getTagName())) {
+            return null;
+        }
+        String path = elem.getAttribute("path");
+        Retain retain = new Retain();
+        retain.setPath(path);
+        return retain;
+    }
+
+    private static Dir getDirectory(Element element) {
+        if (!"dir".equals(element.getTagName())) {
+            return null;
+        }
+        String name = element.getAttribute("name");
+        Dir dir = new Dir();
+        dir.setName(name);
+        Node node = element.getFirstChild();
+        while (node != null) {
```

Comment:
Any reason why you aren't iterating over `element.getChildNodes()`, and instead manually getting the first and then calling iteratively `getNextSibling()`? Happens elsewhere in the PR.

Reply:
`element.getChildNodes()` returns a `NodeList`

And the only way to iterate a `NodeList` is something like this:
```
NodeList tmp = outputDirectories.item(0).getChildNodes();
for (int i = 0; i < tmp.getLength(); i++) {
    Node node = tmp.item(i);
    ...
}
```
Which means it loads all the nodes in memory.
The getFirst / getNext loads nodes as needed.

I don't think that memory would be a problem for the kind of files here.
It just felt a bit cleaner. No a strong preference.

Reply:
Ah, not worth changing. Not more readable, and a minor perf penalty, too.  Guava has `Streams.stream(Iterator)`, which we can't use either because there's no Guava and no Iterator.

The reason I thought to ask is because the code you wrote would be smaller and more readable & maintainable with an FP style via `Stream`.

Reply:
FYI: The CLDR-to-ICU converter depends on CLDR which I believe uses Guava, so Guava should be fine here. However, it sounds like this code is good as is anyway.